src/catalog/lib.rs

```diff
@@ -741,6 +757,10 @@ impl Catalog {
     pub fn dump(&self) -> String {
         serde_json::to_string(&self.by_name).expect("serialization cannot fail")
     }
+
+    pub fn get_sink_suffix(&self) -> &Option<String> {
```

Comment:
Rust convention is to elide the "get" prefix on accessor methods, so this should just be called `sink_suffix`. But see above about instead exposing a `creation_time` and `nonce` method.